### PR TITLE
Add helanal

### PR DESCRIPTION
**Pull request by fiona-naughton**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by fiona-naughton** [comment]
Pinging @IAlibay  if you have time to look, since you might know better than me - I think the builds are failing due to a conflict in numpy versions with MDAnalysis and latest version of gsd (optional dependency of MDAnalysis), but this doesn't appear to be affecting the builds for MDAnalysis itself, so I'm not clear on what's best to do here.

File: recipes/helanal/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+requirements:
+  host:
+    - python >=3.9
+    - setuptools >=61.2
+    - versioningit
+    - pip
+  run:
+    - python >=3.9
+    - mdanalysis >=2.0.0
```
Comment (IAlibay) [inline recipes/helanal/meta.yaml:25]:
```suggestion
    - mdanalysis >=2.0.0
    - gsd <3.4
```

**Review by IAlibay — COMMENTED** [review]
@fiona-naughton I had a look through this and it seems like there's a weird dependency resolution issue with the gsd feedstock. I.e. if I isolate the install to just gsd and try to pin numpy to < 2, it will still pick up the gsd version that requires numpy >= 2.

My temporary solution here is to pin gsd to < 3.4.0, we can unpin later once MDAnalysis has released a new version.

File: recipes/helanal/meta.yaml (hunk 1 above)
Comment (fiona-naughton) [reply recipes/helanal/meta.yaml:25]: kk, thanks! I had a similar thought but wasn't sure if there was a 'nicer' way to resolve it

**Comment by fiona-naughton** [comment]
@conda-forge/help-python, ready for review!

File: recipes/helanal/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{% set name = "helanal" %}
```
Comment (ocefpaf) [inline recipes/helanal/meta.yaml:1]:
```suggestion
{% set python_min = python_min|default("3.10") %}
{% set name = "helanal" %}
```

File: recipes/helanal/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  sha256: b53370d13e5d22d8179c5da9e193af2aa853851e89611241500a906434a7616c
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python >=3.10
```
Comment (ocefpaf) [inline recipes/helanal/meta.yaml:19]:
```suggestion
    - python {{ python_min }}
```

File: recipes/helanal/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    - setuptools >=61.2
+    - versioningit
+    - pip
+  run:
+    - python >=3.10
```
Comment (ocefpaf) [inline recipes/helanal/meta.yaml:24]:
```suggestion
    - python >={{ python_min }}
```

File: recipes/helanal/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    - mdanalysis >=2.0.0
+    - gsd <3.4
+
+test:
+  imports:
+    - helanal
+  commands:
+    - pip check
+  requires:
+    - pip
```
Comment (ocefpaf) [inline recipes/helanal/meta.yaml:34]:
```suggestion
    - pip
    - python {{ python_min }}
```

**Review by ocefpaf — COMMENTED** [review]
You recipe is OK but this syntax I'm adding here is kind of new and we want to use it moving forward.

**Pull request merged**